packages/core/tsconfig.json

```diff
@@ -29,8 +29,20 @@
     "lib": ["dom", "dom.iterable", "esnext"],
     "skipLibCheck": true,
     "esModuleInterop": true,
-    "strict": false
+    "strict": false,
+    "plugins": [
+      {
+        "name": "next"
+      }
+    ],
+    "strictNullChecks": true
```

Comment:
do we need this change? 
I'm not sure if this can impact existing code. 
cc @pedromtec (that was testing some ts configs recently).

Reply:
@emersonlaurentino  This setting is currently breaking the build. Do we need to implement this change right now?

**suggestion**: If not, we can do it afterwards in another PR, since we'll need to fix all the affected occurrences 😬

Reply:
@hellofanny can u share the errors?

Reply:
sure! I found one issue and fixed it, but then I encountered another occurrence. I haven't checked everything yet, so I'm not sure if there are more occurrences.

<img width="784" alt="image" src="https://github.com/user-attachments/assets/e786985d-6e31-431c-b69c-b2941ea3c1c7">

<img width="673" alt="image" src="https://github.com/user-attachments/assets/03c31f0e-7d10-4e2b-bf57-9f17288fa4a5">

Reply:
But I dont change any about that 🤔

Reply:
Hello team. Yes, if you activate the `strictNullChecks` you will receive several errors. Errors that occur because you are accessing an attribute that can be undefined or null. This rule is part of the strict ruleset and generates many problems in this project. I recommend deactivating and creating a task to prioritize this improvement.

Reply:
if you run `yarn build` on root in `main` branch, you'll see that the core package tsconfig file will change... I don't know the reason!

Reply:
build error fixed ✅ 

https://github.com/vtex/faststore/pull/2417#issuecomment-2297784927